jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/completion/HandlerCompletionProvider.kt

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.completion
+
+import com.intellij.codeInsight.completion.CompletionParameters
+import com.intellij.codeInsight.completion.CompletionResultSet
+import com.intellij.codeInsight.lookup.CharFilter
+import com.intellij.openapi.project.Project
+import com.intellij.util.textCompletion.TextCompletionProvider
+import software.aws.toolkits.core.utils.error
+import software.aws.toolkits.core.utils.getLogger
+import software.aws.toolkits.core.utils.info
+import software.aws.toolkits.jetbrains.services.lambda.RuntimeGroup
+import software.aws.toolkits.jetbrains.services.lambda.runtimeGroup
+import java.lang.IllegalStateException
+
+class HandlerCompletionProvider(private val project: Project) : TextCompletionProvider {
+
+    private val logger = getLogger<HandlerCompletionProvider>()
+
+    // Suppress since gradle compiler mismatch return types for property delegate and fail build step
+    private val handlerCompletion: HandlerCompletion? by lazy {
+        val runtimeGroup = RuntimeGroup.determineRuntime(
```

Comment:
I assume this means we only support one type of handler per project - so if they have different source languages then they'd only get one - which is probably fine.

Reply:
@kiiadi, do you mean the case when you have different files with python and java that both define lambda handlers inside one project?

Reply:
yes

Reply:
@kiiadi, could you pleas take a look on a new commit? I've updated the logic to make HanderCompletion runtime dependent. In run configuration, it will be re-created every time new runtime is selected.